subgraphs/_reference_/src/sdk/protocols/generic/pool.ts

```diff
@@ -0,0 +1,387 @@
+    return this.pool.id;
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol supplySideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param rev {BigDecimal} The value to add to the protocol's supplySideRevenue.
+   */
+  addSupplySideRevenueUSD(rev: BigDecimal): void {
+    this.pool.cumulativeTotalRevenueUSD =
+      this.pool.cumulativeTotalRevenueUSD.plus(rev);
+    this.pool.cumulativeSupplySideRevenueUSD =
+      this.pool.cumulativeSupplySideRevenueUSD.plus(rev);
+    this.save();
+
+    this.protocol.addSupplySideRevenueUSD(rev);
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol protocolSideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param rev {BigDecimal} The value to add to the protocol's protocolSideRevenue.
+   */
+  addProtocolSideRevenueUSD(rev: BigDecimal): void {
+    this.pool.cumulativeTotalRevenueUSD =
+      this.pool.cumulativeTotalRevenueUSD.plus(rev);
+    this.pool.cumulativeProtocolSideRevenueUSD =
+      this.pool.cumulativeProtocolSideRevenueUSD.plus(rev);
+    this.save();
+
+    this.protocol.addProtocolSideRevenueUSD(rev);
+  }
+
+  /**
+   * Adds a given USD value to the pool and protocol's supplySideRevenue and protocolSideRevenue. It can be a positive or negative amount.
+   * Same as for the rest of setters, this is mostly to be called internally by the library.
+   * But you can use it if you need to. It will also update the protocol's snapshots.
+   * @param protocolSide {BigDecimal} The value to add to the protocol's protocolSideRevenue.
+   * @param supplySide {BigDecimal} The value to add to the protocol's supplySideRevenue.
+   */
+  addRevenueUSD(protocolSide: BigDecimal, supplySide: BigDecimal): void {
+    this.addSupplySideRevenueUSD(supplySide);
+    this.addProtocolSideRevenueUSD(protocolSide);
+  }
+
+  /**
+   * Convenience method to add revenue denominated in the pool's input token. It converts it to USD
+   * under the hood and calls addRevenueUSD.
+   */
+  addRevenueNative(protocolSide: BigInt, supplySide: BigInt): void {
+    const pricer = this.protocol.pricer;
+    const inputToken = this.getInputToken();
+    const pAmountUSD = pricer.getAmountValueUSD(inputToken, protocolSide);
+    const sAmountUSD = pricer.getAmountValueUSD(inputToken, supplySide);
+    this.addRevenueUSD(pAmountUSD, sAmountUSD);
+  }
+
+  /**
+   * Adds a given amount to the pool's mintSupply. It should only be used for pools of type BURN_MINT.
+   * @param amount {BigInt} The amount to add to the pool's mintSupply. It can be positive or negative.
+   */
+  addMintSupply(amount: BigInt): void {
```

Comment:
Do we have `mintSupply` for generic schema?

Reply:
We do not. Removed.